tests/gtest/avifincrtest_helpers.cc

```diff
@@ -118,13 +118,16 @@ uint32_t GetMinDecodedRowCount(uint32_t height, uint32_t cell_height,
 struct PartialData {
   avifROData available;
   size_t full_size;
+
+  // Only used as non persistent input.
+  std::vector<uint8_t> buffer, swapped_buffer;
```

Comment:
Optional: This approach works, but it may not catch some errors when running under ASan because the buffer returned by the previous io read call is kept around as valid memory in `swapped_buffer`.

Since we can run this test under ASan, a simpler approach is to declare `std::unique_ptr<uint8_t[]> buffer`, allocate a new `buffer` in each io read call, and flip the bits in the old `buffer` (or memset the old `buffer` to 0 or a value like 0xdd) before freeing the old `buffer`.